### DISCUSSION TITLE
How does DAX Studio get a list of open PBIX file Names and their associated port number?

### DISCUSSION THREAD

**Discussion by ejazahmedev** [DaxStudio/DaxStudio#702] [Q&A]
I have an Excel Model that runs a series of DAX Queries on its power-pivot data model and saves csv files using the ADODB Connection. I am planning to port it to PowerBI for good, and the only thing stopping me is the ability to run a series of DAX Queries one after the other and save them to csv files.

I think it would be a good project to have an excel add-in that gives the user the ability to "update" a connection with the latest port number and initial catalog name based on the name of the PBIX file that is open. I love how DAX studio has implemented it. 

While I did find some information on getting a list of port numbers using the 'msmdsrv.port.txt' method, I was not able to figure out how it can be associated with a file name. 

I am not skilled enough to go over your source code. I would appreciate it if you could help me figure out the best way to get a list of open PBIX file names, their ports and their Initial Catalog name.

**Comment by dgosbell** [comment]
> the ability to run a series of DAX Queries one after the other and save them to csv files.

I do have some ideas for some potential future features that might enable this sort of scenario, but I have not started work on anything like this yet.

> While I did find some information on getting a list of port numbers using the 'msmdsrv.port.txt' method, I was not able to figure out how it can be associated with a file name.

So it's not really a simple process. What I do is to: 
* find all the msmdsrv.exe processes running on the machine
* then I get the parent process 
* Then I use a Win32 API call to get the title of the main window of that parent process and strip off the " - Power BI Desktop" from the end of that.

To get the initial catalog you have to first connect to the instance and run a discover request for the DBSCHEMA_CATALOGS DMV

**Reply by liosc** [reply]
Thanks for explaing how you get the pbix names. After you get it, how do you get the database and port linked to that pbix?

**Reply by dgosbell** [reply]
I use a win32 api to get the TCP table and then I get the open port that matches the process id to get port number. The database name does not matter, there is only ever 1 database
